activerecord/lib/active_record/tasks/sqlite_database_tasks.rb

```diff
@@ -38,7 +38,16 @@ def charset
       def structure_dump(filename, extra_flags)
         dbfile = configuration["database"]
         flags = extra_flags.join(" ") if extra_flags
-        `sqlite3 #{flags} #{dbfile} .schema > #{filename}`
+
+        ignore_tables = ActiveRecord::SchemaDumper.ignore_tables
+        if ignore_tables.any?
+          tables = `sqlite3 #{dbfile} .tables`.split - ignore_tables
+          condition = tables.map { |table| "tbl_name = '#{table}'" }.join(" OR ")
```

Comment:
It feels like it might be more consistent, if nothing else, to use a `NOT IN (..)` here, instead of doing the subtraction ourselves.